### PR TITLE
ES-96/SBOM generation

### PR DESCRIPTION
**Pull request by seanbrereton**
Creating SBOM using the cyclonedx plugin.
Adding a task to zip the sboms into 1 folder and publish them.
Tested locally: https://gradle.dev.r3.com/s/mfpjlwzubrjww

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -75,6 +76,18 @@ allprojects {
 
     apply plugin: 'org.jetbrains.kotlin.jvm'
 
+    if (it.toString().contains(":applications:workers:release:")) {
+        apply plugin: 'org.cyclonedx.bom'
+        cyclonedxBom.dependsOn("assemble")
+        cyclonedxBom {
+            includeConfigs = ["runtimeClasspath"]
+            skipProjects = [rootProject.name]
+            skipConfigs = ["integrationTestCompileOnlyDependenciesMetadata", "testCompileOnlyDependenciesMetadata", "compileOnlyDependenciesMetadata", "apiDependenciesMetadata"]
```
Comment (ronanbrowne) [inline build.gradle]: `skipConfigs `should not be needed if `includeConfigs` id used.

File: build.gradle [hunk 2]
```diff
@@ -75,6 +76,18 @@ allprojects {
...
+            destination = file("build/reports/SBOM")
```
Comment (ronanbrowne) [inline build.gradle]:
```suggestion
            destination = file("${buildDIr}/reports/SBOM")
```

**Review by ronanbrowne — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 2 above)
Comment (yarivus) [reply build.gradle]: > 

Actually should be 
```suggestion
            destination = file("${buildDir}/reports/SBOM")
```

File: build.gradle [hunk 3]
```diff
@@ -268,6 +281,39 @@ allprojects {
     }
 }
 
+tasks.register('createSBOMZip', Zip) {
+    description = 'Create SBOM zip file for release'
+    group = 'documentation'
+    String srcDestination = buildDir.toString() + "/archive/SBOM/"
```
Comment (yarivus) [inline build.gradle]: `srcDestination` is confusing ... I propose `sourceDestFolder` or `sourceTargetFolder` (or Folder=>Path)

**Review by yarivus — CHANGES_REQUESTED** [review]

**Review by yarivus — DISMISSED** [review]
LGTM

**Review by ronanbrowne — APPROVED** [review]
LGTM

File: settings.gradle [hunk 4]
```diff
@@ -50,6 +50,7 @@ pluginManagement {
         id "com.jfrog.artifactory" version artifactoryPluginVersion
         id 'io.snyk.gradle.plugin.snykplugin' version snykVersion
         id 'org.jetbrains.dokka' version dokkaVersion
+        id "org.cyclonedx.bom" version "1.7.4"
```
Comment (ronanbrowne) [inline settings.gradle]: Sorry missed this. Can we externalize the property to gradle.properties?

**Review by ronanbrowne — CHANGES_REQUESTED** [review]
See comment

**Review by yarivus — APPROVED** [review]
LGTM

**Review by ronanbrowne — APPROVED** [review]
LGTM

**Pull request merged**